core/keyserver/epochs_test.go

```diff
@@ -70,48 +74,104 @@ func TestGetEpochStream(t *testing.T) {
 	}
 }
 
+type batchStorage map[int64]SourceMap // Map of Revision to Sources
+
+func (b batchStorage) ReadBatch(ctx context.Context, dirID string, rev int64) (*spb.MapMetadata, error) {
+	return &spb.MapMetadata{Sources: b[rev]}, nil
+}
+
+type mutations map[int64][]*mutator.LogMessage // Map of logID to Slice of LogMessages
+
+func (m *mutations) Send(ctx context.Context, dirID string, mutation *pb.EntryUpdate) error {
+	return errors.New("unimplemented")
+}
+
+func (m *mutations) ReadLog(ctx context.Context, dirID string,
+	logID, low, high int64, batchSize int32) ([]*mutator.LogMessage, error) {
+	logShard := (*m)[logID]
+	low = low + 1 // Begin exclusive
```

Comment:
`low++`?

Reply:
No, because it's not an increment in a loop kind of thing